### ISSUE TITLE
Load specific app modules for apps hosted by wwahost.exe

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4569]
**Reported by jteh on 2014-10-23 07:35**
wwahost.exe hosts many Windows Store apps (though it seems some do have their own executables). To allow for application specific customisation, we should identify the specific application in the app module name.

I think we can use GetApplicationUserModelId (in kernel32) for this purpose. This includes a "!" character, so we'll need to check whether Python will cope with this and replace if not.
**Blocked by #4360**
**Blocking #4259**

**Comment by leonardder** [comment]
cc @josephsl

**Comment by josephsl** [comment]
Hi,

This issue mainly affects Windows 8.x, as Windows 10 do come with apps which have their own executables. Note that appPath property from another pull request might help out if needed.

Thanks.

**Comment by josephsl** [comment]
Hi,

Import test: test fails - files with an exclamation mark in the middle of the name isn't imported.

I'd go for command line version (see the example code fragment above), as it can be passed onto app module handler string replacement patch. I'll send a pull request soon.

Thanks.

**Comment by josephsl** [comment]
Hi,

An unexpected case showed up on Windows 10: The Twitter PWA (progressive web app) that is hosted inside WWAHost.exe. If PWA's are hosted inside wwahost, then this boosts priority of this issue, as more PWA's may show up on Microsoft Store.

Thanks.

**Comment by josephsl** [comment]
Hi,

Actually, GetApplicationUserModelId is way faster than command line method, validating @JcSTeh's hunch.

Thanks.

**Comment by josephsl** [comment]
Hi,

PR #9119 has a side effect: the application role workaround for WWAHost breaks, meaning browse mode will be enforced. This means we may need to add a special case for this in IAccessible/MSHTML, or re-examine the app role strategy.

Thanks.
